### PR TITLE
Add smart pointer for internal FBs

### PR DESCRIPTION
**Pull request by mx990**
Add a smart pointer to manage the lifecycle of internal FBs in an FB container.

### CODE REVIEW

File: src/core/fbcontainer.h [hunk 1]
```diff
@@ -155,6 +167,46 @@ namespace forte {
...
+                mFB(std::make_unique<T>(paInstanceNameId, paContainer)) {
+          mFB->configureFB(paConfigString);
+          paContainer.addFB(*mFB);
+        }
+
+        ~CInternalFB() {
+          mFB->getParent().removeFB(*mFB);
+        }
+
+        T *get() noexcept {
```
Comment (azoitl) [inline src/core/fbcontainer.h:190]: do we really need access to the pointer?

File: src/core/fbcontainer.h [hunk 2]
```diff
@@ -155,6 +167,46 @@ namespace forte {
...
+          return mFB.get();
+        }
+
+        T &operator*() noexcept {
+          return mFB.operator*();
+        }
+
+        T *operator->() noexcept {
```
Comment (azoitl) [inline src/core/fbcontainer.h:198]: same

Comment (mx990) [reply src/core/fbcontainer.h:198]: We need the `->` operator for access to members of the wrapped FB class (e.g., `fb_FB1->var_DI1`).

File: src/core/fbcontainer.h (hunk 1 above)
Comment (mx990) [reply src/core/fbcontainer.h:190]: When we do need a direct pointer to the wrapped FB class, this avoids constructs such as `&(*fb_FB1)`.

Comment (azoitl) [reply src/core/fbcontainer.h:190]: make sense

File: src/core/fbcontainer.h (hunk 2 above)
Comment (azoitl) [reply src/core/fbcontainer.h:198]: ah forgot about that.

**Pull request merged**